packages/vscode-extension/src/project/deviceSession.ts

```diff
@@ -167,45 +169,41 @@ export class DeviceSession implements Disposable {
   public async activate() {
     this.isActive = true;
     this.buildManager.activate();
+    this.toolsManager.activate();
   }
 
   public async deactivate() {
     this.isActive = false;
     this.buildManager.deactivate();
+    this.toolsManager.deactivate();
   }
 
-  public async perform(type: ReloadAction): Promise<ReloadResult> {
+  public async perform(type: ReloadAction): Promise<boolean> {
```

Comment:
I think I prefered the `ReloadResult` enum (though with the failure reported via exception, like you do now), but I understand that might be controversial 😆.

You basically need to read this method's implementation to know that `true` means success (ok, that you could probably guess, seeing that we return a boolean from a function which only has side effects), and `false` means "cancelled" (this you have no chance of guessing without looking here)

Reply:
Ideally, I'd throw the `CancelError` from here and rethrow it in the webview (as `CancelError` as well, so that you can tell them apart at call site), but I understand that requires changes to our rpc layer.